### PR TITLE
Updated API calls

### PR DESCRIPTION
**Pull request by MitchellBrownNE**
Added various functions that access the API, along with HTTP, JSON , and other exception handling. 1 second delay between all API calls has been added due to restrictions on API being 1 call / second and was having errors previously.

### CODE REVIEW

**Review by Grewestr — COMMENTED** [review]
looks good

File: api_calls.py [hunk 1]
```diff
@@ -1,81 +1,234 @@
...
+                print("Error: ", response.status, response.reason)
+                return None
+            
+            data = response.read()
+            json_data = json.loads(data.decode("utf-8"))
+
+            # Loop through the games and find the game id the team name
+            for game in json_data['games']:
+                if game['home']['name'] == team_name:
+                    game_id = game['id']
```
Comment (Nobody96103) [inline api_calls.py:38]: Couldn't this line just be 'return game['id']?

File: api_calls.py [hunk 2]
```diff
@@ -1,81 +1,234 @@
...
+                print("Error: ", response.status, response.reason)
+                return None
+            
+            data = response.read()
+            json_data = json.loads(data.decode("utf-8"))
+
+            # Loop through the games and find the team id from param team_name
+            for game in json_data['games']:
+                if game['home']['name'] == team_name:
+                    team_id = game['home']['id']
```
Comment (Nobody96103) [inline api_calls.py:70]: I realize this is nitpicky, but I think this could be replaced with a return statement.

**Review by Nobody96103 — COMMENTED** [review]
Just making sure I have a code review done - I'll see if I can find the one I *actually* reviewed (before I knew how to officially do code review).

Edit: found it - there was never a pull request to main, so it wasn't official: 
https://github.com/WSU-4110/Live-Sports-Display/commit/d513ccf2cce0f1e386619535b0161d165d9dcc16

**Pull request closed without merging**